lualib/resty/events/broker.lua

```diff
@@ -73,19 +71,87 @@ local function broadcast_events(self, unique, data)
     log(DEBUG, "event published to ", n, " workers")
 end
 
+local function read_thread(self, worker_connection)
+    while not terminating(self, worker_connection) do
+        local data, err = worker_connection:recv_frame()
+        if err then
+            if not is_timeout(err) then
+                return nil, "failed to read event from worker: " .. err
+            end
+
+            -- timeout
+            goto continue
+        end
+
+        if not data then
+            if not exiting() then
+                log(ERR, "did not receive event from worker")
+            end
+            goto continue
+        end
+
+        local event_data, err = decode(data)
+        if not event_data then
+            if not exiting() then
+                log(ERR, "failed to decode event data: ", err)
+            end
+            goto continue
+        end
+
+        -- unique event
+        local unique = event_data.spec.unique
+        if unique then
+            if self._uniques:get(unique) then
+                log(DEBUG, "unique event is duplicate: ", unique)
```

Comment:
We can add to the message "skip it" or something.

Reply:
Yes, we can, but I didn't change the log messages.

Reply:
I think that now we have no reason to change the message, it is not the point of this PR.